### ISSUE TITLE
Do not symlink files in please location

### ISSUE THREAD

**Issue by sagikazarmark** [thought-machine/please#1256]
The `pleasew` script (and maybe `plz` too) symlinks the downloaded files to the `~/.please` directory and it's not immediately obvious why.

Adding the correct version to the PATH (eg. `~/.please/15.5.0`) should work equally well.

As for the please binary (I'm not even sure it checks downloaded versions, but I guess it does) we can do the same.

Is there a reason for this?

(I'm working on a GitHub action for Please and the symlinking makes things a bit harder)

**Comment by Tatskaari** [comment]
Once #1192 is done this will work a lot better. The current idea as to how this should work is you install please into `~/.please/{vesrion}` and it will symlink the active version to `~/please`. You are then meant to add `~/.please` to the path.

As you point out this isn't really that nice as stray symlinks or broken links can bork your please installation. 

The new recommended approach will be to put the `sudo cp pleasew /usr/bin/plz` or the like.

**Comment by sagikazarmark** [comment]
I think the notion of "active version" doesn't really make sense, I'd rather call it "default" version. If you pin the version (which should be a best practice IMHO) in `.plzconfig` it should be used (without "switching" the active version).

If the version is not pinned, it should always use the latest version. How the latest version is detected is a good question. I'd probably update the "active version" when a new version of please is downloaded.

tl;dr: I wouldn't update the active version when using a pinned version.

> If you want to pre-load the please versions for ci, just unzip the tarbals into

That doesn't work at the moment, `pleasew` tries to download a new version when it doesn't find a symlinked `please` binary.

**Comment by Tatskaari** [comment]
> I think the notion of "active version" doesn't really make sense, I'd rather call it "default" version. If you pin the version (which should be a best practice IMHO) in `.plzconfig` it should be used (without "switching" the active version).

Oh, interesting idea. @peterebden what do you think about this? 

> If the version is not pinned, it should always use the latest version. How the latest version is detected is a good question. I'd probably update the "active version" when a new version of please is downloaded.

We could just use https://get.please.build/latest_version

> tl;dr: I wouldn't update the active version when using a pinned version.
> 
> > If you want to pre-load the please versions for ci, just unzip the tarbals into
> 
> That doesn't work at the moment, `pleasew` tries to download a new version when it doesn't find a symlinked `please` binary.


Good points! I'll keep this in mind when I get round to this. Will let you know what I come up with if you're interested!

**Comment by sagikazarmark** [comment]
> Will let you know what I come up with if you're interested!

Sure! I want to keep my please action updated: https://github.com/sagikazarmark/setup-please-action

**Comment by Tatskaari** [comment]
I will track this work in #1192
